test/control/LoadHandlerTest.cpp

```diff
@@ -303,6 +311,141 @@ class LoadHandlerTest : public CppUnit::TestFixture
+
+			Stroke* e4 = (Stroke*) (*layer->getElements())[4];
+			CPPUNIT_ASSERT_EQUAL(ELEMENT_STROKE, e4->getType());
+
+			Text* e5 = (Text*) (*layer->getElements())[5];
+			CPPUNIT_ASSERT_EQUAL(ELEMENT_TEXT, e5->getType());
+
+			Stroke* e6 = (Stroke*) (*layer->getElements())[6];
+			CPPUNIT_ASSERT_EQUAL(ELEMENT_STROKE, e6->getType());
+
+			Stroke* e7 = (Stroke*) (*layer->getElements())[7];
+			CPPUNIT_ASSERT_EQUAL(ELEMENT_STROKE, e7->getType());
+
+			return elements;
+		};
+		LoadHandler handler;
+		Document* doc1 = handler.loadDocument(GET_TESTFILE("packaged_xopp/suite.xopp"));
+		auto elements1 = getElements(doc1);
+
+		SaveHandler h;
+		h.prepareSave(doc1);
+		Path tmp = Util::getTmpDirSubfolder() / "save.xopp";
+		h.saveTo(tmp);
+
+		// Create a second loader so the first one doesn't free the memory
+		LoadHandler handler2;
+		Document* doc2 = handler2.loadDocument(tmp.str());
+		auto elements2 = getElements(doc2);
+
+		// Check that the coordinates from both files don't differ more than the precision they were saved with
+		auto coordEq = [](double a, double b) { return std::abs(a - b) <= 1e-8; };
+
+		for (int i = 0; i < elements1->size(); i++)
+		{
+			Element* a = elements1->at(i);
+			Element* b = elements2->at(i);
+			CPPUNIT_ASSERT_EQUAL(a->getType(), b->getType());
+			CPPUNIT_ASSERT(coordEq(a->getX(), b->getX()));
+			CPPUNIT_ASSERT(coordEq(a->getY(), b->getY()));
+			CPPUNIT_ASSERT(coordEq(a->getElementWidth(), b->getElementWidth()));
+			CPPUNIT_ASSERT(coordEq(a->getElementHeight(), b->getElementHeight()));
+			CPPUNIT_ASSERT_EQUAL(a->getColor(), b->getColor());
+			switch (a->getType())
+			{
+			case ELEMENT_STROKE:
+			{
+				Stroke* sA = dynamic_cast<Stroke*>(a);
+				Stroke* sB = dynamic_cast<Stroke*>(b);
+				CPPUNIT_ASSERT_EQUAL(sA->getPointCount(), sB->getPointCount());
+				CPPUNIT_ASSERT_EQUAL(sA->getToolType(), sB->getToolType());
+				CPPUNIT_ASSERT_EQUAL(sA->getLineStyle().hasDashes(), sB->getLineStyle().hasDashes());
+				CPPUNIT_ASSERT(coordEq(sA->getAvgPressure(), sB->getAvgPressure()));
+				for (int i = 0; i < sA->getPointCount(); i++)
+				{
+					Point pA = sA->getPoint(i);
+					Point pB = sB->getPoint(i);
+					CPPUNIT_ASSERT(coordEq(pA.x, pB.x));
+					CPPUNIT_ASSERT(coordEq(pA.y, pB.y));
+					CPPUNIT_ASSERT(coordEq(pA.z, pB.z));
+				}
+				break;
+			}
+			case ELEMENT_TEXT:
+			{
+				Text* tA = dynamic_cast<Text*>(a);
+				Text* tB = dynamic_cast<Text*>(b);
+				CPPUNIT_ASSERT_EQUAL(tA->getText(), tB->getText());
+				CPPUNIT_ASSERT_EQUAL(tA->getFont().getSize(), tB->getFont().getSize());
+				break;
+			}
+			default:
+				// If other elements are to be used in the test, implement extra comparisons
+				CPPUNIT_ASSERT(false);
+				break;
+			}
+		}
+	}
+
+#ifdef __linux__
+	void testLoadStoreLoadGerman()
+	{
+		constexpr auto testLocale = "de_DE.UTF-8";
+		bool hasTestLocale = false;
+		for (auto available = g_get_language_names(); *available; available++)
```

Comment:
I've already got de_DE.utf8 installed yet g_get_language_names() only gave a subset of locales (which didn't include de_DE.UTF-8)
`locale -a`  on the command line verified it was installed.
Commenting out 426: CPPUNIT_ASSERT(hasTestLocale); resulted in a sucessful test.
inserting:
```
char* currLocale = setlocale(LC_ALL, testLocale);
        g_warning("curr Locale = %s",currLocale);
```
verified that the locale was being loaded ( it returns null for failed setLocale )
`** (process:29428): WARNING **: curr Locale = de_DE.UTF-8`
so...
 It may be better to just set the locale and assert the return string from setLocale() != null ??

Reply:
Thanks for requesting the review.. forced me to get aquainted with the testing apparatus .. which is something I should've done when I first started contributing. Thanks also for adding this test!

Reply:
great suggestion